src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.TestLibrary/System.Text.Json.TestLibrary.Roslyn4.4.csproj

```diff
@@ -3,6 +3,6 @@
   <Import Project="System.Text.Json.TestLibrary.targets" />
 
   <ItemGroup>
-    <ProjectReference Include="..\..\gen\System.Text.Json.SourceGeneration.Roslyn4.0.csproj" OutputItemType="Analyzer" ReferenceOutputAssembly="false" />
+    <ProjectReference Include="..\..\gen\System.Text.Json.SourceGeneration.Roslyn4.4.csproj" OutputItemType="Analyzer" ReferenceOutputAssembly="false" />
```

Comment:
You don't want to move the existing roslyn4.0 test projects to roslyn4.4 as otherwise you would lose test coverage for the roslyn4.0 compiled source generator. Instead, you want to add roslyn4.4 test projects. We also still test the roslyn3.11 roslyn source generator as we live build it.

Reply:
It's a good point. I've been primarily motivated by https://github.com/dotnet/runtime/issues/68353 when making this change -- adding yet another test target felt like it might impact editor performance even more even though the sourcegen components are identical. But I guess we might need to have to bite the bullet on this one, given that we will need to keep supporting 4.0. cc @ericstj

Reply:
> https://github.com/dotnet/runtime/issues/68353

Very interesting!

We do have at least one other source generator that targets multiple roslyn versions. I would prefer to build on-top of the current infrastructure and be consistent in regards to testing source generators across the repo than to make this one special. That said, as long as we test all various versions of the source generators, I'm fine with any solution.

Aside from that but related, we should come up with lifecycle policy for source generators. I imagine that it's likely that we add yet another System.Text.Json source generator that targets a future version of roslyn4.X during .NET 9. Keeping all of these building and consumable doesn't sound like a good future-proof plan. I.e. our runtime / TFM support policy allows us to drop TFMs that are out-of-support, i.e. net5.0. We should come-up with a similar strategy for source generators. Would you be interesting in driving such a conversation? I think that others already tried gathering feedback but never drove this topic to a decision point.

Reply:
I'll revert the commit that changes the tested version, to focus this PR on NRT infrastructure.